packages/commerce-sdk-react/src/hooks/useMutation.ts

```diff
@@ -85,18 +96,61 @@ export const useCustomMutation = (
                     }
                 },
                 clientConfig: {
-                    parameters: {
-                        clientId: config.clientId,
-                        siteId: config.siteId,
-                        organizationId: config.organizationId,
-                        shortCode: config.organizationId
-                    },
-                    proxy: config.proxy,
+                    ...currentClientConfig,
                     ...clientConfig
                 }
             })
         }
     }
 
-    return useReactQueryMutation(callCustomEndpointWithAuth(apiOptions), mutationOptions)
+    const callCustomEndpointWithBody = async (args: {
+        body: unknown
+        parameters?: {[key: string]: string | number | boolean | string[] | number[]}
+        headers?: {[key: string]: string}
+    }) => {
+        const clientConfig = apiOptions.clientConfig || {}
+        const {access_token} = await auth.ready()
+        return await helpers.callCustomEndpoint({
+            ...apiOptions,
+            options: {
+                ...apiOptions.options,
+                headers: {
+                    Authorization: `Bearer ${access_token}`,
+                    ...clientHeaders,
+                    ...apiOptions.options.headers,
+                    ...args.headers
+                },
+                body: args.body,
+                parameters: {
+                    ...apiOptions.options.parameters,
+                    ...args.parameters
+                }
+            },
+            clientConfig: {
+                ...currentClientConfig,
+                ...clientConfig
+            }
+        })
+    }
+
+    if (!apiOptions?.options?.body) {
+        // If users don't define a body when they use this hook, they can pass in a body later
+        // when calling mutate() or mutateAsync()
+        // this allows users to call the same endpoint with different arguments
+
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        return useReactQueryMutation(
+            callCustomEndpointWithBody as MutationFunction<unknown, unknown>,
```

Comment:
why do we have to explicitly cast the type here?

Reply:
This was the bit that threw me off for a long time. The reason we have to cast it is because the other optional return has this type and they need to match. Good luck inferring that from the TS error output though

Reply:
And re-thinking this would trigger an extensive "re-think" with a big ripple effect throughout the very rigid type system across the library

Reply:
looks like with out casting it is resolving to an incorrect overloaded type.

Reply:
That's correct, we have this large tech debt with huge piles of overloaded types that no one understands. It's very tragic, really